Sources/AsyncAlgorithms/AsyncThrottleSequence.swift

```diff
@@ -81,7 +81,16 @@ extension AsyncThrottleSequence: AsyncSequence {
       let start = last ?? clock.now
       repeat {
         guard let element = try await base.next() else {
-          return nil
+          if reduced != nil, let last {
+            // ensure the rate of elements never exceeds the given interval
+            let amount = interval - last.duration(to: clock.now)
+            if amount > .zero {
+              try? await clock.sleep(for: amount)
```

Comment:
Why are we swallowing the cancellation error here?

Reply:
Wondering if we should throw in the case where we are a throwing sequence and if not just return `nil`.

Reply:
We can't really know that - plus throwing `CancellationError` seems rather off to me except in the cases where EOF and cancellation need disambiguation

Reply:
It is impossible to differentiate when a non-throwing async sequence returns `nil` if it was due to EOF or task cancellation, that's why I am leaning towards always throwing the `CancellationError` if possible. Now this case is interesting because we would just return the reduced value and probably the next iteration will potentially throw the `CancellationError`. So I guess it's fine